docs/tasks/logs.rst

```diff
@@ -0,0 +1,259 @@
+format. Generally, these logs can be categorized into *system logs* and
+*datapath logs*.
+
+System logs are logs emitted by the Kanister to track important controller
+events like interactions with the Kubernetes APIs, CRUD operations on
+blueprints and actionsets etc.
+
+Datapath logs, on the other hand, are logs emitted by data mover task pods
+created by Kanister. These logs are streamed to the Kanister controller before
+the task pods are terminated to ensure they are not lost inadvertently. Datapath
+log lines usually include the ``LogKind`` field, with its value set to
+``datapath``.
+
+The rest of this documentation provides instructions on how to segregate
+Kanister's system logs from datapath logs using Loki_ and Grafana_.
+
+To run the provided commands, access to a Kubernetes cluster using the
+``kubectl`` and ``helm`` command-line tools is required.
+
+Follow the instructions in the :ref:`helm` page to deploy Kanister on the
+cluster.
+
+Deployments Setup
+=================
+
+The commands and screenshots in this documentation are tested with the following
+software versions:
+
+* Loki 2.5.0
+* Grafana 8.5.3
+* Promtail 2.5.0
+
+.. image:: img/logs-kanister-loki-grafana.png
+
+Let's begin by installing Loki. Loki is a datastore optimized for holding log
+data. It indexes log data via streams made up of logs, where each stream is
+associated with an unique set of labels.
+
+.. code-block:: bash
+
+  helm repo add grafana https://grafana.github.io/helm-charts
+
+  helm repo update
+
+  helm -n loki install --create-namespace loki grafana/loki \
+    --set image.tag=2.5.0
+
+Confirm that the Loki StatefulSet is successfully rolled out:
+
+.. code-block:: bash
+
+  kubectl -n loki rollout status sts/loki
+
+.. code-block:: bash
+
+  statefulset rolling update complete 1 pods at revision loki-6d7b88b95c...
+
+.. note::
+  The Loki configuration used in this installation is meant for demonstration
+  purposes only. The Helm chart deploys a non-HA single instance of Loki,
+  managed by a StatefulSet workload. See the `Loki installation documentation`_
+  for other installation methods that may be more suitable for your
+  requirements.
+
+Use Helm to install Grafana with a pre-configured Loki data source:
+
+.. code-block:: bash
+
+  svc_url=$(kubectl -n loki get svc loki -ojsonpath='{.metadata.name}.{.metadata.namespace}:{.spec.ports[?(@.name=="http-metrics")].port}')
+
+  cat <<EOF | helm -n grafana install --create-namespace grafana grafana/grafana -f -
+  datasources:
+    datasources.yaml:
+      apiVersion: 1
+      datasources:
+      - name: Loki
+        type: loki
+        url: http://$svc_url
+        access: proxy
+        isDefault: true
+  EOF
+
+Confirm that the Grafana Deployment is successfully rolled out:
+
+.. code-block:: bash
+
+  kubectl -n grafana rollout status deploy/grafana
+
+.. code-block:: bash
+
+  deployment "grafana" successfully rolled out
```

Comment:
can we have these two in the same code block?

Reply:
I removed the second code block, as they don't seem necessary.